juriscraper/opinions/united_states/state/nytrial.py

```diff
@@ -0,0 +1,223 @@
+        if citation:
+            metadata["Citation"] = citation
+
+        return metadata
+
+    def get_docket_number(self, scraped_text: str) -> str:
+        """Get docket number
+        Sometimes it is explicit in a table at the beginning of the document
+        with the heading 'Docket Number'
+
+        Sometimes it is explicit in the first lines of the text with headings
+        such as 'Index No', 'Docket No', etc
+
+        Sometimes it is just a special string, that may be numeric only
+        Sometimes it may not exist
+
+        :param scraped_text: scraped text
+        :return: docket number if it exists
+        """
+        if "<table" in scraped_text:
+            html = html_fromstring(scraped_text)
+            docket_xpath = "//table[@width='75%']/following-sibling::text()"
+            element = html.xpath(docket_xpath)
+            docket = element[0] if element else ""
+        else:
+            docket_regex = r"Docket Number:(?P<docket>.+)"
+            match = re.search(docket_regex, scraped_text[:500])
+            docket = match.group("docket") if match else ""
+
+        return docket.strip()
+
+    def get_judge(self, scraped_text: str) -> str:
+        """Get judge from PDF or HTML text
+
+        We delete a trailing ", J." or ", S." after judge's last names,
+        which appear in HTML tables. This are honorifics
+        For "S.": https://www.nycourts.gov/reporter/3dseries/2023/2023_50144.htm
+
+        :param scraped_text: string from HTML or PDF
+        :return: judge name
+        """
+        if "<table" in scraped_text:
+            html = html_fromstring(scraped_text)
+            td = html.xpath(
+                '//td[contains(text(), ", J.") or contains(text(), ", S.")]'
+            )
+            judge = td[0].text_content() if td else ""
+        else:
+            match = re.search(r"Judge:\s?(?P<judge>.+)", scraped_text)
+            judge = match.group("judge") if match else ""
+
+        judge = normalize_judge_string(clean_string(judge))[0]
+        judge = re.sub(r" [JS]\.$", "", judge)
+
+        return judge.strip()
+
+    def get_citation(self, scraped_text: str) -> Dict[str, str]:
+        """Extracts volume, reporter and page of citation that
+        has the following shape [81 Misc 3d 1211(A)]
+
+        Citation should be searched on the top of the document since an opinion
+        may cite other opinions on the argumentation
+
+        Tagged as "Official Citation" on the source. For example:
+        https://lrb.nycourts.gov/citator/reporter/citations/detailsview.aspx?id=2023_51315
+
+        :param scraped_text: string from HTML or PDF
+        :return: dictionary with expected citation fields
+        """
+        regex = r"\[(?P<volume>\d+) (?P<reporter>Misc 3d) (?P<page>.+)\]"
+        match = re.search(regex, scraped_text[:1200])
+
+        if not match:
+            return {}
+
+        return match.groupdict("")
```

Comment:
I think we can drop this code if we use the regex above - but this is all very nice code.

Reply:
Great, I agree with you that simple is better and regex should be enough in this case. I will test the changes and make a new push